pkg/api/v1alpha1/types.go

```diff
@@ -38,6 +38,9 @@ type DeschedulerPolicy struct {
 	// EvictLocalStoragePods allows pods using local storage to be evicted.
 	EvictLocalStoragePods *bool `json:"evictLocalStoragePods,omitempty"`
 
+	// EvictDaemonSetPods allows pods owned by a DaemonSet resource to be evicted.
+	EvictDaemonSetPods *bool `json:"evictDaemonSetPods,omitempty"`
```

Comment:
If we want this to work for v1alpha1, then the conversion has to be updated to include it [ref](https://github.com/kubernetes-sigs/descheduler/blob/bb5930eb21605879104e02d9da62bd96180d9492/pkg/api/v1alpha1/conversion.go#L115)

Reply:
To be fair I believe this can be a dangerous behavior, but making it by default disabled and allowing enabling it optionally seems to be a good common ground.